### PR TITLE
[BEAM-13925] Add weekly automation to update our reviewer config

### PR DESCRIPTION
**Pull request by damccorm**
Right now, we don't have a well defined automated system for staying on top of pull request reviews - we rely on contributors being able to find the correct OWNERS file and committers manually triaging/calling attention to old pull requests. This is the continuation of an effort to improve our automation around this. [Design doc for the full effort here](https://docs.google.com/document/d/1FhRPRD6VXkYlLAPhNfZB7y2Yese2FCWBzjx67d3TjBo/edit#).

This specific change adds automation to automatically find and suggest reviewers for different areas **and** remove inactive reviewers from the configuration

I mostly tested this locally to make sure it came up with the correct configuration on this repo, I also did test that it correctly creates the pr here - https://github.com/damccorm/beam-pr-bot-demo/pull/11
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @KonradJanica for TS changes

File: scripts/ci/pr-bot/shared/reviewerConfig.ts [hunk 1]
```diff
@@ -22,10 +22,24 @@ import { Label } from "./githubUtils";
...
   constructor(pathToConfigFile) {
     this.config = yaml.load(
       fs.readFileSync(pathToConfigFile, { encoding: "utf-8" })
     );
+    this.configPath = pathToConfigFile;
+  }
+
+  // Returns all possible reviewers for each label configured.
+  getReviewersForAllLabels(): { [key: string]: string[] } {
+    var labelObjects = this.config.labels;
```
Comment (KonradJanica) [inline scripts/ci/pr-bot/shared/reviewerConfig.ts:35]: Avoid `var`.

**Review by KonradJanica — APPROVED** [review]

**Review by damccorm — COMMENTED** [review]
@youngoli for final review!

**Review by lostluck — APPROVED** [review]
LGTM Thanks!

**Pull request merged**